File: myCTCA Mobile/iOS/src/myctca/Models/ROI.swift
//
//  ROI.swift
//  myctca
//
//  Created by Tomack, Barry on 2/5/18.
//  Copyright © 2018 CTCA. All rights reserved.
//

import Foundation

/**
 This object was declared as a struct to make it smaller and copyable instead of passing by reference.
    1. Encapsulates a few relatively simple data values.
    2. Reasonable to expect that the encapsulated values will be copied rather than referenced
    3. Properties stored by the struct are themselves value types
    4. The struct does not need to inherit properties or behavior from another existing type.
 Struct instances are allocated on stack, and class instances are allocated on heap so sructs are faster.
 
 Codable means that it can be directly encoded (for JSON, in this case)
 */

struct ROI : Codable {
    
    var selectedFacility: String?
    var firstName: String?
    var lastName: String?
    var dateOfBirth: String?
    var selectedDeliveryMethod: [String] = [String]()
    var pickupDate: String?
    var selectedAuthorizationAction = [String]()
    var facilityOrIndividual: String?
    var address: String?
    var city: String?
    var state: String?
    var zip: String?
    var phoneNumber: String?
    var fax: String?
    var emailAddress: String?
    var selectedPurposes: [String] = [String]()
    var beginningOfTreatment: Bool = false
    var beginDate: String?
    var endOfTreatment: Bool = false
    var endDate: String?
    var restrictions: String?
    var selectedDisclosureInformation: [String] = [String]()
    var disclosureInformationOther: String?
    var selectedHighlyConfidentialDiscolosureInformation: [String] = [String]()
    var signature: String?
    var patientRelation: String?
    
    init() { }
    
}

File: myCTCA Mobile/Android/src/myctca/app/src/main/java/com/myctca/model/ROI.java
package com.myctca.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by tomackb on 2/5/18.
 */

public class ROI {

    public String selectedFacility;
    public String firstName;
    public String lastName;
    public String dateOfBirth;
    public List<String> selectedDeliveryMethod = new ArrayList<>();
    public String pickupDate;
    public List<String> selectedAuthorizationAction = new ArrayList<>();
    public String facilityOrIndividual;
    public String address;
    public String city;
    public String state;
    public String zip;
    public String phoneNumber;
    public String fax;
    public String emailAddress;
    public List<String> selectedPurposes = new ArrayList<>();
    public boolean beginningOfTreatment = false;
    public String beginDate;
    public boolean endOfTreatment = false;
    public String endDate;
    public String restrictions;
    public List<String> selectedDisclosureInformation = new ArrayList<>();
    public String disclosureInformationOther;
    public List<String> selectedHighlyConfidentialDiscolosureInformation = new ArrayList<>();
    public String signature;
    public String patientRelation;

}
